src/components/application_manager/src/app_service_manager.cc

```diff
@@ -528,4 +532,256 @@ void AppServiceManager::AppServiceUpdated(
+                      "PassThroughRequests");
+        // Construct list of pass through services
+        request_queue_[correlation_id] = std::queue<RpcPassThroughRequest>();
+        auto services = app_service_manager_.GetActiveServices();
+
+        for (auto services_it = services.begin(); services_it != services.end();
+             ++services_it) {
+          auto handled_rpcs =
+              services_it
+                  ->record[strings::service_manifest][strings::handled_rpcs];
+          for (size_t i = 0; i < handled_rpcs.length(); i++) {
+            if (handled_rpcs[i].asInt() == function_id) {
+              // Add Passtrhough requests to queue
+              rpc_message[strings::params][strings::connection_key] =
+                  services_it->connection_key;
+              auto req = RpcPassThroughRequest{origin_connection_key,
+                                               services_it->connection_key,
+                                               rpc_message};
+              request_queue_[correlation_id].push(req);
+              app_manager_.IncreaseForwardedRequestTimeout(
+                  origin_connection_key, correlation_id);
+              break;
+            }
+          }
+        }
+
+        rpc_message[strings::params][strings::connection_key] =
+            origin_connection_key;
+        auto req = RpcPassThroughRequest{
+            origin_connection_key, origin_connection_key, rpc_message};
+        request_queue_[correlation_id].push(req);
+
+        LOG4CXX_DEBUG(logger_,
+                      "Added " << request_queue_[correlation_id].size()
+                               << " requests to the queue");
+        CycleThroughRPCPassingRequests(correlation_id);
+
+      } else {
+        LOG4CXX_DEBUG(logger_, "Correlation id does exist in map. Continuing");
+        return false;
+      }
+
+      if (request_queue_[correlation_id].empty()) {
+        LOG4CXX_DEBUG(logger_,
+                      "No services left in map. using core to handle request");
+        request_queue_.erase(correlation_id);
+        current_request_.erase(correlation_id);
+        messages_using_core_.insert(correlation_id);
+        return false;
+      } else {
+        LOG4CXX_DEBUG(
+            logger_,
+            "Forwarding request to service with app id "
+                << current_request_[correlation_id].destination_connection_key);
+        smart_objects::SmartObjectSPtr result =
+            std::make_shared<smart_objects::SmartObject>(
+                current_request_[correlation_id].message);
+        AddTimeoutTimer(correlation_id);
+        app_manager_.GetRPCService().SendMessageToMobile(result);
+        return true;
+      }
+
+    } break;
+
+    case MessageType::kResponse: {
+      LOG4CXX_DEBUG(logger_, "Handle response");
+      if (!rpc_message[strings::msg_params].keyExists(strings::result_code)) {
+        break;
+      }
```

Comment:
To clear out existing timers (so they don't fire), shouldn't you just call `RemoveTimeoutTimer` here, rather than inside of other cases?

Reply:
Fixed 👍